sdk/src/Core/Amazon.Runtime/Internal/Transform/JsonErrorResponseUnmarshaller.cs

```diff
@@ -92,8 +87,7 @@ public ErrorResponse Unmarshall(JsonUnmarshallerContext context)
                 }
 
                 // strip extra data from type, leaving only the exception type name
-                type = type == null ? null : type.Substring(type.LastIndexOf("#", StringComparison.Ordinal) + 1);
-
+                type = type == null ? null : ParseType(type.Substring(type.LastIndexOf("#", StringComparison.Ordinal) + 1));
```

Comment:
If type != null and if code is null or empty it is going to call ParseType on type twice. That seems it would produce the wrong type value in that case. Or is it only passing because type wouldn't have another `:` in it?

Reply:
Hmm, if there was another `:` before the actual error, then it would return the wrong error type. But, in that case the first ParseType would return the wrong error anyways. 

I added the second ParseType here because if both the `__type` and header don't specify the error, type is populated in the if block (83-86), in which case it is the full type (i.e. `aws.protocoltests.restjson#FooError:http://internal.amazon.com/coral/com.amazon.coral.validate/`). In this case we need to call ParseType to get the error. 

The only case I can think of this returning the wrong error type is if there is a `:` in the the namespace, but the docs specify: 

> If a : character is present, then take only the contents before the first : character in the value.

so either way we just need to blindly take the index of the first `:`